97.md

```diff
@@ -0,0 +1,251 @@
+* Defines a new tag "attachment"
+* Uses the binary messages in the websocket connection
+
+
+New tag "attachment"
+-------------
+The proposal introduces a new tag  "attachment" with the following format
+
+```
+["attachment","hash_hex","size_in_bytes","mime_type","features1", "feature2", ....]
+```
+
+* **hash_hex** - hexadecimal representation of SHA256 hash of the attached binary file. This string also serves as **attachment-id**
+* **size_in_bytes** - size of the attachment in bytes
+* **mime_type** - mime type of the attachment 
+* **features...** - optional fields with various features in format `<field>=<value>`
+    * **blurhash=** - contains NIP-94 blurhash
+    * **purpose=** - purpose of the attachment in context of current note. This can help client
+to select optimal rendering. Suggested purposes
+        * **thumbnail** - thubnail image of a following video
+        * **story** - all attachments marked by this purpose are played as a story
+        * **inline** - attachment is rendered in text, referenced by `#[<index>]`. This purpose hides this attachment from the list of attachments in note's detail.
+
+```
+["attachment","12887...","64000","image/jpeg","blurhash=qwdjq3...", "purpose=thumbnail"]
+["attachment","874788...","2258147","video/mpeg4", "purpose=story"]
+["attachment","ae3c758...","2158799","video/mpeg4", "purpose=story"]
+["attachment","788eq78...","254123","image/jpeg", "purpose=story"]
+```
+        
+
+* One event can have more than one of these tags, then it carries more attachments.
+* Each relay defines the maximum attachment size and also the maximum number of attachments within an event
+* Attachment can be attached to any "kind". Attachments for `kind:4` are always encrypted
+
+The presence of the `attachment` attribute only informs the client that the relay should also have its own attachment content available in binary form. The client must download the attachments from the same relay to display them. See below
+
+
+Upload
+------
+Single event must be published simultaneously with attachments. It is necessary to prevent the client from publishing the event but not uploading the relevant attachments. Similarly, it should not be possible to upload attachments first and then publish the event that references them. The operation must be atomic
+
+For this reason, a new command is introduced: "ATTACH"
+
+**Protocol flow**
+
+```
+client: ["ATTACH", <event>]
+relay:  ["OK","<event-id">,true,"continue"]
+client: <binary message for first attachment>
+relay:  ["ATTACH",<attachment-id>, true, "continue"]
+client: <binary message for second attachment>
+relay:  ["ATTACH",<attachment-id>, true, "continue"]
+...
+...
+client: <binary message for last attachment>
+relay:  ["ATTACH",<attachment-id>, true, "complete"]
+     //event is published
+```
+
+The protocol flow is designed in such a way that it is possible to post other commands between individual phases. Possible implementation on a relay:
```

Comment:
This rule is important because it allows you to send e.g. [REQ] commands at the same time as the upload. At the same time, you cannot influence whether the relay itself sends you an event in the meantime as a response to your REQ. It's just important to mention that the requests don't have to follow each other. That's why each answer contains the identifier of what it answers

Reply:
Agree, my point is that concurrent filters and sends are the default behavior of Nostr. There is no need to specify it. 

You only have the need to specify because you are creating a temporary atomic transaction substate in each submission, which doesn't exist anywhere else in the Nostr protocol.